tests/CommonGateway/CoreBundle/Service/NotificationServiceTest.php

```diff
@@ -0,0 +1,235 @@
+        $this->syncService = $this->createMock(SynchronizationService::class);
+        $this->resourceService = $this->createMock(GatewayResourceService::class);
+
+        $this->notificationService = new NotificationService(
+            $this->entityManager,
+            $this->logger,
+            $this->syncService,
+            $this->resourceService
+        );
+    }
+
+    /**
+     * Tests the notificationHandler with get method.
+     *
+     * @return void
+     */
+    public function testNotificationHandler_WithGetMethod_ReturnsData()
+    {
+        // Arrange
+        $data = ['method' => 'GET'];
+        $configuration = [];
+
+        // Act
+        $result = $this->notificationService->notificationHandler($data, $configuration);
+
+        // Assert
+        $this->assertSame($data, $result);
+    }
+
+    /**
+     * Tests the notificationHandler with post method finding and syncing an object.
+     *
+     * @return void
+     */
+    public function testNotificationHandler_WithPostMethod_FindsSyncAndSynchronizesObject()
+    {
+        // Arrange
+        $data = [
+            'method'      => 'POST',
+            'url'         => 'https://example.com/api/v1/objects/1'
+        ];
+        $configuration = [
+            'entity' => 'https://example.com/example.schema.json',
+            'urlLocation' => 'url',
+        ];
+        $synchronization = $this->createMock(Synchronization::class);
+        $source          = $this->createMock(Source::class);
+        $schema          = $this->createMock(Entity::class);
+
+
+        $this->resourceService->expects($this->once())
+            ->method('findSourcesForUrl')
+            ->with($data['url'])
+            ->willReturn([$source]);
+
+        $this->resourceService->expects($this->once())
+            ->method('getSchema')
+            ->with($configuration['entity'])
+            ->willReturn($schema);
+
+        $this->syncService->expects($this->once())
+            ->method('findSyncBySource')
+            ->with($source, $schema, '1')
+            ->willReturn($synchronization);
+
+        $this->syncService->expects($this->once())
+            ->method('synchronize')
+            ->with($synchronization);
+
+        $this->entityManager->expects($this->once())
+            ->method('flush');
+
+        $expectedResponse = new Response(json_encode(['Message' => 'Notification received, object synchronized']), 200, ['Content-type' => 'application/json']);
+        $data['response'] = $expectedResponse;
+
+        // Act
+        $result = $this->notificationService->notificationHandler($data, $configuration);
+
+        // Assert
+        $this->assertEquals($expectedResponse->getContent(), $result['response']->getContent());
+        $this->assertEquals($expectedResponse->getStatusCode(), $result['response']->getStatusCode());
+        $this->assertSame($expectedResponse->headers->all(), $result['response']->headers->all());
```

Comment:
Will this test work with the used $data and $configuration?
Because doing ->notificationHandler() will try to actually get from a source and sync information.
So the expected response here is probably not a 200 OK 🤔

Reply:
That's where the mocks are for: we pretend that the sync will return a 200 OK